### PR TITLE
Add the end-to-end test for Auditor mode

### PR DESCRIPTION
**Pull request by supl**
This PR adds the end-to-end test for Auditor mode.

There's no logic change so could be reviewed quickly.

### CODE REVIEW

File: node/common/client_service_base.ts [hunk 1]
```diff
@@ -223,7 +223,8 @@ export class ClientServiceBase {
           else resolve();
         });
       });
-      return this.executePromise(promise) as Promise<void>;
+
+      await this.executePromise(promise);
```
Comment (supl) [inline node/common/client_service_base.ts:227]: This is a miss.

**Review by ypeckstadt — APPROVED** [review]
LGTM. Thank you.

**Review by jnmt — APPROVED** [review]
LGTM! Thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**